manifests/agent.pp

```diff
@@ -151,10 +151,13 @@
   validate_string($agent_package_name)
   validate_string($agent_service_name)
 
-  if ( ( $ossec_ip == undef ) and ( $ossec_hostname == undef ) and ( $ossec_address == undef ) ) {
-    fail('must pass either $ossec_ip or $ossec_hostname or $ossec_address to Class[\'wazuh::agent\'].')
+  if (($manage_client_keys == 'yes')){
+      if ( ( $ossec_registration_ip == undef ) or ( $ossec_reporting_ip == undef ) ) {
```

Comment:
Here, we have a `or`,  but if we look below, this class cannot work without a value for `ossec_registration_ip`.

We should change the condition or adapt the code below.

Reply:
Oh, you are completely right, I will apply the fixes on it!

Thanks for reporting!